### PR TITLE
feat(home): build initial React components

### PR DESCRIPTION
**Pull request by wescopeland**
This PR is the first of **four** PRs that will migrate the home page to Inertia.js / React:

PR 1 of 4: (This PR) Build some basic components with hardcoded data. Lots of functionality missing, no back-end.
PR 2 of 4: Use real data from the back-end / Laravel controllers rather than hardcoded data.
PR 3 of 4: Rebuild the Active Players component.
PR 4 of 4: Finalize the swap of the PHP home page for the React home page.

---

The intent of this PR is to build basic components for the home page that contain hardcoded data and extremely simple code coverage. This serves as a foundation to actually rebuild the page.

**Lots of pieces of functionality are still missing which will arrive in subsequent PRs.**

The React page is available at `/demo/home`:

Screenshot 2024-10-21 at 8 51 31 PM

I will describe each component in brief.

### News
* The carousel is gone.
* Three news posts and a "See more" link (to a currently non-existent page) are visible.

Screenshot 2024-10-21 at 8 52 41 PM

Screenshot 2024-10-21 at 8 53 43 PM

### New Sets/Revisions
* Renamed to "Just Released".
* Virtually identical with the existing Blade UI.

Screenshot 2024-10-21 at 8 54 37 PM

Screenshot 2024-10-21 at 8 54 47 PM

### Active Players
* Shows a static empty state at the moment. This component will be rebuilt in PR 3 of 4. **This will have an entire PR dedicated towards it**, as it is by far the most complex component on the page.

### Currently Online
* No longer uses Google Charts. Now uses Recharts/D3.js.
* Shows the all-time recorded high (currently hardcoded, but it's available in the logs in server storage).

https://github.com/user-attachments/assets/93fa3f60-16d3-45fb-9de6-632991b11d73

### Sets in Progress
* Renamed to "Latest Sets in Progress".
* Virtually identical with the existing Blade UI.

Screenshot 2024-10-21 at 8 58 23 PM

Screenshot 2024-10-21 at 8 58 32 PM

### Forum Activity
* Renamed to "Forum Recent Posts".
* Virtually identical with the existing Blade UI.

Screenshot 2024-10-21 at 8 59 55 PM

Screenshot 2024-10-21 at 9 00 08 PM

### Top Links
* Uses similar iconography for points as introduced in #2758.
* Is much more responsive at tablet breakpoints.

Screenshot 2024-10-21 at 9 01 01 PM

Screenshot 2024-10-21 at 9 01 20 PM

### Achievement of the Week
* Uses a more minimal design.

Screenshot 2024-10-21 at 9 02 25 PM

### Statistics
* Redesigned.
* Changes the unlocks label to "Achievement Unlocks".
* Is much more responsive at tablet breakpoints.

Screenshot 2024-10-21 at 9 02 51 PM

Screenshot 2024-10-21 at 9 03 18 PM

### Recent Game Awards
* Minor CSS changes.
* Is much more responsive at tablet breakpoints.

Screenshot 2024-10-21 at 9 04 21 PM

Screenshot 2024-10-21 at 9 04 14 PM

### CODE REVIEW

File: resources/js/features/home/components/+root/FrontPageNews/FrontPageNews.tsx [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      <div className="grid grid-cols-2 gap-4 sm:flex sm:flex-col sm:gap-1">
+        <NewsCard
+          authorDisplayName="SporyTike"
+          href="#"
+          imageSrc="https://media.retroachievements.org/Images/103356.jpg"
+          postedAt="2d ago"
+          title="New Guide: Dragon Warrior II"
+          lead="A new achievement guide has been published for Dragon Warrior II, written by Nepiki! Continue the Quest and fulfill the legend."
+          tagLabel="Guide"
+        />
```
Comment (Jamiras) [inline resources/js/features/home/components/+root/FrontPageNews/FrontPageNews.tsx:25]: I'm not fond of the gradient shadow on the images for these. I can't find the component in this PR, so I'm not sure if it's something we can change.

File: lang/en_US.json [hunk 2]
```diff
@@ -308,6 +308,51 @@
...
+    "All-time High: :number (:date)": "All-time High: :number (:date)",
+    "Welcome!": "Welcome!",
+    "Build your profile, track your progress, and compete with friends on your favorite classic games.": "Build your profile, track your progress, and compete with friends on your favorite classic games.",
+    "We provide": "We provide",
+    "the emulators": "the emulators",
+    ", you just need": ", you just need",
+    "the games": "the games",
+    ". From Atari 2600 to PlayStation 2, and everything in between.": ". From Atari 2600 to PlayStation 2, and everything in between.",
+    "Which of these achievements do you think you can get?": "Which of these achievements do you think you can get?",
+    "Forum Recent Posts": "Forum Recent Posts",
```
Comment (Jamiras) [inline lang/en_US.json:355]: This reads weird. Suggest either "Recent Forum Posts" or "Forum - Recent Posts"

File: resources/js/features/home/components/+sidebar/TopLinks/TopLinks.tsx [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+  return (
+    <div className="flex flex-col gap-2.5">
+      <a href={route('download.index')} className={baseButtonVariants({ size: 'sm' })}>
+        <LuDownload className="mr-2 h-4 w-4 text-sky-400" />
+        {t('Download Emulator')}
+      </a>
+
+      <div className="flex flex-col gap-2.5 sm:grid sm:grid-cols-2 lg:flex">
+        <a href="/globalRanking.php" className={baseButtonVariants({ size: 'sm' })}>
+          <LuCircleDot className="mr-2 h-4 w-4 text-amber-400" />
```
Comment (Jamiras) [inline resources/js/features/home/components/+sidebar/TopLinks/TopLinks.tsx:32]: > Uses similar iconography for points as introduced in 2758

Screenshots there have the medal for points and the circle for progress. Are these backwards?

File: resources/js/features/home/components/+root/FrontPageNews/FrontPageNews.tsx (hunk 1 above)
Comment (wescopeland) [reply resources/js/features/home/components/+root/FrontPageNews/FrontPageNews.tsx:25]: Looks like I accidentally forgot to turn that off. Updated in latest.

It only makes sense if there's a tag in the image corner, like this:

Screenshot 2024-10-27 at 11 40 41 AM

It's coming from `<NewsCard />`.

Helpful tip: VSCode has very good built-in React editing and debugging support. To trace any React component, on Windows, hold down the Control key and then click on the component name in the template. VSCode will take you right to the component's definition.

File: lang/en_US.json (hunk 2 above)
Comment (wescopeland) [reply lang/en_US.json]: Updated in latest.

File: resources/js/features/home/components/+sidebar/TopLinks/TopLinks.tsx (hunk 3 above)
Comment (wescopeland) [reply resources/js/features/home/components/+sidebar/TopLinks/TopLinks.tsx]: Good catch. I did indeed get these icons backwards. Fixed in latest.

**Review by Jamiras — APPROVED** [review]

**Comment by wescopeland** [comment]
Additional note related to home page news: I have recently made changes in https://github.com/RetroAchievements/RAWeb/pull/2572 that are sufficient enough that it should be re-reviewed.

**Comment by wescopeland** [comment]
I once again had to blow away our entire lockfile due to a bug in npm:
Screenshot 2024-10-30 at 6 27 47 PM

This issue isn't our fault and defeats the entire purpose of having a lockfile.

Now that our package.json file is beginning to stabilize, I'm going to work on migrating us over to pnpm.

**Pull request merged**